goship.go

```diff
@@ -760,23 +791,216 @@ func extractOutputHandler(fn func(http.ResponseWriter, *http.Request, string, st
 	}
 }
 
+type User struct {
+	UserName, UserAvatar string
+	auth                 auth
+}
+
+func getUser(r *http.Request) (User, error) {
+	u := User{}
+	if authentication.authorization != true {
+		u.UserName = *defaultUser
+		u.UserAvatar = *defaultAvatar
+		return u, nil
+	}
+	session, err := store.Get(r, sessionName)
+	if err != nil {
+		return u, errors.New("Error Getting User Session")
```

Comment:
@ramen74 we can define errors as package vars so that they only have to be allocated once